after_success.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash -u
+# -*- indent-tabs-mode: nil  -*-
+
+# Software License Agreement - BSD License
+#
+# This script runs after successful travis run
+#
+# Author:  Tyler Weaver
+
+# Helper functions
+source .travis/util.sh
+
+if [[ "${TEST:=}" == *code-coverage* ]]; then
+  echo -e $(colorize BOLD "Generating codecov.io report")
+  travis_run --title "codecov.io report upload" bash <(curl -s https://codecov.io/bash)
```

Comment:
This is almost too easy ;)

Reply:
It was too easy, I discovered that reports weren't actually being generated because the build directory was still owned by root and not the travis user.  I'm updating this with one more command to fix this.